EventListener/BreadcrumbListener.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace WhiteOctober\BreadcrumbsBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+use Symfony\Component\Routing\RouterInterface;
+use WhiteOctober\BreadcrumbsBundle\Attribute\Breadcrumb;
+use WhiteOctober\BreadcrumbsBundle\Model\Breadcrumbs;
+
+class BreadcrumbListener
+{
+    private ControllerArgumentsEvent $event;
+    private PropertyAccessor $propertyAccess;
+
+    public function __construct(
+        private readonly Breadcrumbs     $breadcrumbs,
+        private readonly RouterInterface $router,
+    )
+    {
+        $this->propertyAccess = PropertyAccess::createPropertyAccessor();
+    }
+
+    public function onKernelController(ControllerArgumentsEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+        $this->event = $event;
+
+        /** @var Breadcrumb $breadcrumbAttribute */
+        foreach ($event->getAttributes(Breadcrumb::class) as $breadcrumbAttribute) {
+            $text = $breadcrumbAttribute->getText();
+
+            if (empty($text)) {
+                $this->addBreadcrumb($breadcrumbAttribute);
+                continue;
+            }
+
+            $pattern = '/\{[^}]+\}/';
+            preg_match_all($pattern, $text, $matches, PREG_SET_ORDER, 0);
```

Comment:
Wouldn't it be better to use translation feature for this instead of having this whole logic for replacing placeholders in strings here?

Reply:
Could you clarify which translation function you're referring to? I'm not quite sure what you mean.

Reply:
@mhujer ping )))